### DISCUSSION TITLE
Strange issue/bug with QDate mask?

### DISCUSSION THREAD

**Discussion by jimmiejackson414** [quasarframework/quasar#12896] [General - Components / Directives / etc]
I have a QDate component with range option, that I'm attempting to create a custom mask with using [DayJS](https://day.js.org/docs/en/durations/diffing).

When I look at the computed property being generated for `formattedDatepicker` in the Vue Devtools, it returns "7 Days", however what is being displayed on the page in the input mask is a misspelled "7 dfys". Any idea what could cause this??

```vue
<template>
  <q-input
    v-model="stringifiedDate"
    outlined
    dense
    :mask="formattedDatepicker">
    <template #prepend>
      <q-icon
        name="bi-calendar-check"
        class="cursor-pointer">
        <q-popup-proxy
          ref="qDateProxy"
          cover
          transition-show="scale"
          transition-hide="scale">
          <q-date
            v-model="data.datepicker"
            range>
            <div class="row items-center justify-between">
              <q-btn
                v-close-popup
                label="Close"
                flat
                color="text"
                :ripple="false" />
                <q-btn
                  v-close-popup
                  label="Select"
                  :ripple="false"
                  color="primary"
                  flat />
              </div>
            </q-date>
        </q-popup-proxy>
      </q-icon>
    </template>
  </q-input>
</template>

<script lang='ts' setup>
  import {
    computed, reactive, ref, watch,
  } from 'vue';
  import dayjs from 'dayjs';
  import duration from 'dayjs/plugin/duration';
  import relativeTime from 'dayjs/plugin/relativeTime';

  dayjs.extend(duration);
  dayjs.extend(relativeTime);

  const data = reactive({
    datepicker: {
      from: undefined as string | undefined,
      to: undefined as string | undefined,
    },
  });

  const stringifiedDate = computed(() => JSON.stringify(data.datepicker));
  const formattedDatepicker = computed(() => {
    const from = dayjs(data.datepicker.from);
    const to = dayjs(data.datepicker.to);
    return dayjs.duration(from.diff(to))
      .humanize();
  });

  // set local data when dialog is opened
  watch(() => props.isOpen, () => {
    if (props.isOpen) {
      data.datepicker = {
        from: props.trip?.starting_date ? dayjs(props.trip.starting_date)
          .format('YYYY/MM/DD') : undefined,
        to: props.trip?.ending_date ? dayjs(props.trip.ending_date)
          .format('YYYY/MM/DD') : undefined,
      };
    }
  });
</script>
```

**Comment by MilosPaunovic** [comment]
Input mask has its own formatting tokens, as shown here [https://quasar.dev/vue-components/input#mask](https://quasar.dev/vue-components/input#mask). Lowercase letter `a`, contained in a word `days`, is one of 'em.

The simplest solution would be to escape it in a returned string, like this:

```js
const formattedDatepicker = computed(() => {
    const from = dayjs(data.datepicker.from);
    const to = dayjs(data.datepicker.to);
    return dayjs.duration(from.diff(to)).humanize().replaceAll('a', '\\a');
  });
```

But I would suggest that you create helper which will escape all of the mask tokens, if they ever show up.

**Reply by pdanpdan** [reply]
For QDate/QTime mask the escaping is by enclosing the escaped things in square brackets.
https://quasar.dev/vue-components/date#example--mask-with-escaped-characters

**Reply by jimmiejackson414** [reply]
@MilosPaunovic @pdanpdan Fascinating, I had no idea Quasar even included Date Utils like this. 

I'm still unsure of how to escape the characters using the square brackets though... right now it's literally just outputting "[8 dfys]".

Using a `.replaceAll()` appears to be working, though when I change the dates so that it should output "A month", I am instead returned "a month". I think ideally I'd go the first route using the built-in mask escaping method...?
